api_tests/registrations/views/test_registration_detail.py

```diff
@@ -442,6 +454,109 @@ def test_read_write_contributor_cannot_update_custom_citation(
         assert res.status_code == 403
 
 
+@pytest.mark.django_db
+class TestRegistrationWithdrawal(TestRegistrationUpdateTestCase):
+
+    @pytest.fixture
+    def public_payload(self, public_registration, make_payload):
+        return make_payload(
+            id=public_registration._id,
+            attributes={'withdrawn': True, 'withdrawal_justification': 'Not enough oopmh.'}
+        )
+
+    def test_withdraw_registration_fails(
+            self, app, user, read_write_contributor, public_registration, make_payload,
+            private_registration, public_url, private_url, public_project, public_payload):
+        # test withdrawal with no auth
+        res = app.put_json_api(public_url, public_payload, expect_errors=True)
+        assert res.status_code == 401
+
+        # test withdrawal from a read write contrib
+        res = app.put_json_api(public_url, public_payload, auth=read_write_contributor.auth, expect_errors=True)
+        assert res.status_code == 403
```

Comment:
`read_write_contributor` is not actually a contributor on the public_registration, just the private_registration, which is why this test isn't properly testing that write contributors shouldn't be able to withdraw.